runtime/neo4jclient/registry.go

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 Hypermode Inc.
+ * Licensed under the terms of the Apache License, Version 2.0
+ * See the LICENSE file that accompanied this code for further details.
+ *
+ * SPDX-FileCopyrightText: 2024 Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package neo4jclient
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/hypermodeinc/modus/lib/manifest"
+	"github.com/hypermodeinc/modus/runtime/manifestdata"
+	"github.com/hypermodeinc/modus/runtime/secrets"
+	"github.com/neo4j/neo4j-go-driver/v5/neo4j"
+)
+
+var n4j = newNeo4jRegistry()
+
+type neo4jRegistry struct {
+	sync.RWMutex
+	neo4jDriverCache map[string]neo4j.DriverWithContext
+}
+
+func newNeo4jRegistry() *neo4jRegistry {
+	return &neo4jRegistry{
+		neo4jDriverCache: make(map[string]neo4j.DriverWithContext),
+	}
+}
+
+func CloseDrivers(ctx context.Context) {
+	n4j.Lock()
+	defer n4j.Unlock()
+
+	removed := make([]string, 0)
+
+	for key, driver := range n4j.neo4jDriverCache {
+		driver.Close(ctx)
+		removed = append(removed, key)
+	}
+
+	for _, key := range removed {
+		delete(n4j.neo4jDriverCache, key)
+	}
+}
+
+func (nr *neo4jRegistry) getDriver(ctx context.Context, n4jName string) (neo4j.DriverWithContext, error) {
+	nr.Lock()
+	defer nr.Unlock()
+
+	if driver, ok := nr.neo4jDriverCache[n4jName]; ok {
+		return driver, nil
+	}
+
+	for name, info := range manifestdata.GetManifest().Connections {
+		if name != n4jName {
+			continue
+		}
```

Comment:
`Connections` is a map.  You don't have to iterate it - you can just fetch the value from the map directly.  ie.,

```go
man := manifestdata.GetManifest()
info, ok := man.Connections[n4jName]
...
```

Reply:
Not sure why, but the same is being done in `dgraphclient`, so update in both places please.